School leaver positive destinations.R

```diff
@@ -1,59 +1,251 @@
+##########################################################
+
+
+# Function to prepare final files: main_data and popgroup
+prepare_final_files <- function(ind){
+  
+  # 1 - main data (ie data behind summary/trend/rank tab)
+  main_data <- all_data %>% 
+    filter(split_name == "None") %>% 
+    select(code, ind_id, year, 
+           numerator, rate, upci, lowci, 
+           def_period, trend_axis) %>%
+    unique() 
+  
+  write.csv(main_data, paste0(data_folder, "Test Shiny Data/", ind, "_shiny.csv"), row.names = FALSE)
+  write_rds(main_data, paste0(data_folder, "Test Shiny Data/", ind, "_shiny.rds"))
+  # save to folder that QA script accesses:
+  write_rds(main_data, paste0(data_folder, "Data to be checked/", ind, "_shiny.rds"))
+  
+  # 2 - population groups data (ie data behind population groups tab)
+  pop_grp_data <- all_data %>% 
+    filter(!(split_name %in% c("None", "Deprivation (SIMD)"))) %>% 
+    select(code, ind_id, year, numerator, rate, upci, 
+           lowci, def_period, trend_axis, split_name, split_value,) 
+  
+  # Save
+  write.csv(pop_grp_data, paste0(data_folder, "Test Shiny Data/", ind, "_shiny_popgrp.csv"), row.names = FALSE)
+  write_rds(pop_grp_data, paste0(data_folder, "Test Shiny Data/", ind, "_shiny_popgrp.rds"))
+  # save to folder that QA script accesses: (though no QA for popgroups files?)
+  write_rds(pop_grp_data, paste0(data_folder, "Data to be checked/", ind, "_shiny_popgrp.rds"))
+  
+  # 3 - SIMD data (ie data behind deprivation tab)
+  
+  # Process SIMD data
+  simd_data <- all_data %>% 
+    filter(split_name == "Deprivation (SIMD)") %>% 
+    unique() %>%
+    select(-split_name) %>%
+    rename(quintile = split_value) %>%
+    mutate(quint_type = "sc_quin")
+  
+  # Save intermediate SIMD file
+  write_rds(simd_data, file = paste0(data_folder, "Prepared Data/", ind, "_shiny_depr_raw.rds"))
+  write.csv(simd_data, file = paste0(data_folder, "Prepared Data/", ind, "_shiny_depr_raw.csv"), row.names = FALSE)
+  
+  # Run the deprivation analysis (saves the processed file to 'Data to be checked')
+  analyze_deprivation_aggregated(filename = paste0(ind, "_shiny_depr"), 
+                                 pop = "depr_pop_schoolleavers", # the population file created above
+                                 ind_id = 13010, 
+                                 ind_name = ind
+  )
+  
+  # Make data created available outside of function so it can be visually inspected if required
+  main_data_result <<- main_data
+  pop_grp_data_result <<- pop_grp_data
+  simd_data_result <<- simd_data
+  
+  
+}
+
+
+# Run function to create final files
+prepare_final_files(ind = "pos_school_leaver_destinations")
+
+
+# Run QA reports
+####################
+
+# main data: 
+run_qa(filename = "pos_school_leaver_destinations")
+
+# ineq data: # NOT RUNNING DUE TO MISSING HBS???
+run_ineq_qa(filename = "pos_school_leaver_destinations")
+
+
+#END
 
-analyze_first(filename = "school_leaver_destinations", measure = "percent", 
```

Comment:
the previous version of the script ran the numerator and denominator through the analysis functions. This meant that although the source data only contained CA data the functions enable addition of NHS board data (because it could map CA to NHS boards). Ideally we should try and present/create data for as many geography levels as we can. Because i think we have numerator and denominator we can achieve good enough NHS board figures so I think we need to revist this script